modules/unsupported/jdbc-ng/jdbc-hana/src/main/java/org/geotools/data/hana/metadata/CsvReader.java

```diff
@@ -0,0 +1,125 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2018, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.data.hana.metadata;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A simple CSV file reader tailored for reading the embedded HANA spatial metadata.
+ *
+ * @author Stefan Uhrig, SAP SE
+ */
+public class CsvReader {
```

Comment:
Classes in the metadata package are probably best kept package private, with the exception of MetadataImport, they are not really meant to be reused outside this context.